stackit/internal/services/cdn/distribution/resource.go

```diff
@@ -0,0 +1,612 @@
+		return
+	}
+	tflog.Info(ctx, "CDN distribution deleted")
+}
+
+func (r *distributionResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, core.Separator)
+
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error importing CDN distribution", fmt.Sprintf("Expected import identifier on the format: [project_id]%q[distribution_id], got %q", core.Separator, req.ID))
+	}
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), idParts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("distribution_id"), idParts[1])...)
+	tflog.Info(ctx, "CDN distribution state imported")
+}
+
+func mapFields(distribution *cdn.Distribution, model *Model) error {
+	if distribution == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+
+	if distribution.ProjectId == nil {
+		return fmt.Errorf("Project ID not present")
+	}
+
+	if distribution.Id == nil {
+		return fmt.Errorf("CDN distribution ID not present")
+	}
+
+	id := *distribution.ProjectId + core.Separator + *distribution.Id
+
+	model.ID = types.StringValue(id)
+	model.DistributionId = types.StringValue(*distribution.Id)
+	model.ProjectId = types.StringValue(*distribution.ProjectId)
+	model.Status = types.StringValue(*distribution.Status)
+	model.CreatedAt = types.StringValue(distribution.CreatedAt.String())
+	model.UpdatedAt = types.StringValue(distribution.UpdatedAt.String())
+
+	distributionErrors := []attr.Value{}
+	if distribution.Errors != nil {
+		for _, e := range *distribution.Errors {
+			distributionErrors = append(distributionErrors, types.StringValue(*e.En))
+		}
+	}
+	modelErrors, diags := types.ListValue(types.StringType, distributionErrors)
+	if diags.HasError() {
+		return core.DiagsToError(diags)
+	}
+	model.Errors = modelErrors
+
+	regions := []attr.Value{}
+	for _, r := range *distribution.Config.Regions {
+		regions = append(regions, types.StringValue(string(r)))
+	}
+	modelRegions, diags := types.ListValue(types.StringType, regions)
+	if diags.HasError() {
+		return core.DiagsToError(diags)
+	}
+	originRequestHeaders := types.MapNull(types.StringType)
+	if distribution.Config.Backend.HttpBackend.OriginRequestHeaders != nil && len(*distribution.Config.Backend.HttpBackend.OriginRequestHeaders) > 0 {
+		headers := map[string]attr.Value{}
+		for k, v := range *distribution.Config.Backend.HttpBackend.OriginRequestHeaders {
+			headers[k] = types.StringValue(v)
+		}
+		mappedHeaders, diags := types.MapValue(types.StringType, headers)
+		originRequestHeaders = mappedHeaders
+		if diags.HasError() {
+			return core.DiagsToError(diags)
+		}
+	}
+	// note that httpbackend is hardcoded here as long as it is the only available backend
```

Comment:
Stylistic nitpick: A helper variable would make this a bit mor readable:

```suggestion
	if origHeaders := distribution.Config.Backend.HttpBackend.OriginRequestHeaders; origHeaders != nil && len(*origHeaders) > 0 {
		headers := map[string]attr.Value{}
		for k, v := range *origHeaders {
			headers[k] = types.StringValue(v)
		}
		mappedHeaders, diags := types.MapValue(types.StringType, headers)
		originRequestHeaders = mappedHeaders
		if diags.HasError() {
			return core.DiagsToError(diags)
		}
	}
	// note that httpbackend is hardcoded here as long as it is the only available backend
```